File: semestr-8/Programowanie obiektowe w Java/alexander/Java/2020/IZ06TC2/BaseControlTest/src/AnimalFabric.java
public class AnimalFabric 
{
    private static final String[] MAP=new String[] {"Cat","Fly"};
    public static String[] kinds() { return MAP; }
    public static Animal make(String kind,String name)
    {
        switch(kind)
        {
            case "Cat": return new Cat(name);
            case "Fly": return new Fly(name);
        }
        return null;
    }    
}

class Animal
{
    private String name;
    public Animal(String name) { this.name=name; }
    public String path() { return ""; }
    @Override public String toString() { return path() + name; }    
}

class Mammal extends Animal
{
    public Mammal(String name) { super(name); }
    @Override public String path() { return super.path() + "Mammal : "; }
}

class Insect extends Animal
{
    public Insect(String name) { super(name); }
    @Override public String path() { return super.path() + "Insect : "; }
}

class Cat extends Mammal
{
    public Cat(String name) { super(name); }
    @Override public String path() { return super.path() + "Cat : "; }
}

class Fly extends Insect
{
    public Fly(String name) { super(name); }
    @Override public String path() { return super.path() + "Fly : "; }
}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2020/ID06IO1/BuilderAndFabric/BuilderAndFabric/AnimalFabric.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BuilderAndFabric
{
    delegate Animal AnimalMaker();
    class AnimalFabric
    {
        List<KeyValuePair<string, AnimalMaker>> list = new List<KeyValuePair<string, AnimalMaker>>()
        {
            new KeyValuePair<string, AnimalMaker>("Cat", () => new Cat()),
            new KeyValuePair<string, AnimalMaker>("Fly", () => new Fly()),
        };
        //*///
        private static Dictionary<string, AnimalMaker> map = new Dictionary<string, AnimalMaker>()
        {
            {"Cat", () => new Cat() },
            {"Fly", () => new Fly() },
        };
        /*///
        private static Dictionary<string,AnimalMaker> map=makemap();
        private static Dictionary<string, AnimalMaker> makemap()
        {
            Dictionary<string, AnimalMaker> map = new Dictionary<string, AnimalMaker>();
            map.Add("Cat", () => new Cat());
            map.Add("Fly", () => new Fly());
            return map;
        }
        //*///
        public static List<string> keys()
        {
            List<string> lst = map.Keys.ToList();
            lst.Sort();
            return lst;
        }
        public static Animal make(String kind)
        {
            return map[kind]();
        }
    }
    class Animal
    {

    }

    class Cat : Animal
    {
        public override string ToString()
        {
            return "Puszek";
        }
    }

    class Fly : Animal
    {
        public override string ToString()
        {
            return "Bzzz";
        }
    }
}
